modules/renter/contractor/persist.go

```diff
@@ -80,6 +75,17 @@ func (c *Contractor) load() error {
 		}
 		c.currentPeriod = highestEnd - c.allowance.Period
 	}
+
+	// Old perist may need information from the hostdb to fill in missing
+	// fields.
+	allHosts := c.hdb.AllHosts()
+	hmap := make(map[modules.NetAddress]modules.HostDBEntry)
+	// Iterate so that in the event of duplicate hosts for a netaddress, the
+	// highest score host is the one in the map.
+	for _, host := range allHosts {
+		hmap[host.NetAddress] = host
+	}
```

Comment:
it's not too terrible since it is only relevant once: when upgrading to 1.1.1. Might impact LTS though. Worst-case, we could iterate through the whole blockchain again, gathering all the netaddresses.